src/main/java/gr8pefish/ironbackpacks/block/BlockBackpack.java

```diff
@@ -0,0 +1,255 @@
+package gr8pefish.ironbackpacks.block;
+
+
+import gr8pefish.ironbackpacks.IronBackpacks;
+import gr8pefish.ironbackpacks.api.IronBackpacksAPI;
+import gr8pefish.ironbackpacks.api.backpack.BackpackInfo;
+import gr8pefish.ironbackpacks.api.backpack.variant.BackpackVariant;
+import gr8pefish.ironbackpacks.core.RegistrarIronBackpacks;
+import gr8pefish.ironbackpacks.network.GuiHandler;
+import net.minecraft.block.BlockHorizontal;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.properties.IProperty;
+import net.minecraft.block.properties.PropertyDirection;
+import net.minecraft.block.properties.PropertyEnum;
+import net.minecraft.block.state.BlockStateContainer;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityFurnace;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.IItemHandler;
+
+import javax.annotation.Nullable;
+import java.util.Random;
+
+/**
+ * Block with a TileEntity
+ *
+ * {@link TileEntityBackpack}
+ */
+public class BlockBackpack extends AbstractBlockTileEntity<TileEntityBackpack> {
+
+    /** Field for use in unlocalized and registry naming*/
+    public static final String BACKPACK_NAME = "backpack_block";
+
+    /** Field for holding backpack variant data */
+    public BackpackVariant backpackVariant;
+
+    // No longer needed, see the field above
+//    public static final IProperty<BackpackVariantEnum> BACKPACK_VARIANT = PropertyEnum.create("backpack_variant", BackpackVariantEnum.class);
+
+    /** Field for the IProperty */ //TODO: Move to API
+    public static final PropertyDirection FACING = BlockHorizontal.FACING;
+
+    /** Constructor */
+    public BlockBackpack(BackpackVariant backpackVariant) {
+        super(Material.ROCK, BACKPACK_NAME+"_"+backpackVariant.toString().toLowerCase());
+        this.backpackVariant = backpackVariant;
+        setDefaultState(this.blockState.getBaseState().withProperty(FACING, EnumFacing.NORTH)); //Default facing arbitrary
+    }
+
+    //Getter
+
+    public BackpackVariant getBackpackVariant() {
+        return backpackVariant;
+    }
+
+    //TileEntity Methods
+
+    @Override
+    public Class<TileEntityBackpack> getTileEntityClass() {
+        return TileEntityBackpack.class;
+    }
+
+    @Nullable
+    @Override
+    public TileEntityBackpack createTileEntity(World world, IBlockState state) {
+        return new TileEntityBackpack(placeholder); //TODO: BackpackInfo from the itemStack
```

Comment:
To clarify: So the default constructor with no data, then on the block's placement we include the BackpackInfo?

Reply:
Yes. On block placement it would look something like

```java
world.setBlockState(pos, RegistrarIronBackpacks.BACKPACK_BLOCK);
TileEntityBackpack tileBackpack = new TileEntityBackpack();
tileBackpack.setBackpackInfo(backpackInfo);
world.setTileEntity(pos, tileBackpack);
```